### PR TITLE
Implement Smart Cropping feature, fixes #129

### LINKED ISSUE #129: Feature request: Focal point
I've worked in a closed-source version of a software that does pretty much the same as this. One very useful feature that it had was the ability to use an arbitrary point to crop an image to. In flyimg this is sort of achieved with the gravity parameter but it is not very precise. In the one I worked with you could specify an x and y focal point float that ranged from 0 to 1. 

In the context of my use case, this served two main purposes:

1. The user would upload an image and choose the focal point. No matter how we cropped the image, the "important" part (aka focal point) would always be in the center.
2. We ran some images through face detection and set the focal point automatically to the center of the face. Then, cropped profile pictures would always look great even if a user uploaded an image with a weird aspect ratio.

### PR DESCRIPTION
**Pull request by sadok-f**
A very wanted feature: Smart Cropping.
I added a python script to determine to best coordinates for cropping.

### CODE REVIEW

File: python/smartcrop.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/env python
+# Author: https://github.com/davidfischer-ch
+# From Repo: https://github.com/smartcrop/smartcrop.py
+
```
Comment (baamenabar) [inline python/smartcrop.py:4]: This file has a ton of config hardcoded in it. Wouldn't you want to be able to fiddle with these values a bit?

**Review by baamenabar — APPROVED** [review]
Awesome PR!
I'm missing the docs update.

This is a true or false flag, right? Or does it have to go with some other option?

File: python/smartcrop.py (hunk 1 above)
Comment (sadok-f) [reply python/smartcrop.py:4]: Good point, we could later extract them if we need to change the smart crop command line.
for now, I suggest to keep it here and see what feedback we can gather from users using it and adapt what we can configure.

Comment (sadok-f) [reply python/smartcrop.py:4]: Good catch about the docs!
I need to update that 👍

**Pull request merged**